src/Storage/Field/Collection/FieldCollection.php

```diff
@@ -115,7 +116,23 @@ protected function doInitialize()
                 $field = $this->em->getFieldManager()->getFieldFor($fieldtype);
                 $type = $field->getStorageType();
                 $typeCol = 'value_' . $type->getName();
-                $val->setValue($val->$typeCol);
+
+                // Because there's a potential for custom fields that use json storage to 'double hydrate' this causes
+                // json_decode to throw a warning. Here we prevent that by replacing the error handler.
+                set_error_handler(
+                    function ($errNo, $errStr, $errFile) {},
```

Comment:
@rossriley do you have an easily reproducible test case for the double hydration?  The Symfony debug handlers in 3.2 can convert the error into an exception and we should get ready to catch that.